src/services/identity/LaunchClient.ts

```diff
@@ -35,21 +40,133 @@ class LaunchClient {
+        autoSubDomainCreationPatterns: [],
+        autoSubDomainIAMRole: undefined,
+        certificateVerificationDNSRecord: undefined,
+        domainAssociationArn: `arn:aws:amplify:ap-southeast-1:11111:apps/${input.appId}/domains/${input.domainName}`,
+        domainName: input.domainName,
+        domainStatus: "CREATING",
+        enableAutoSubDomain: false,
+        statusReason: undefined,
+        subDomains: undefined,
+      },
+    }
+
+    const subDomainSettingsList = input.subDomainSettings
+    if (!subDomainSettingsList || subDomainSettingsList.length === 0) {
+      return Promise.resolve(mockResponse)
+    }
+
+    const subDomains: SubDomain[] = this.getSubDomains(subDomainSettingsList)
+
+    // We know that domainAssociation is not undefined, so we can use the non-null assertion operator
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    mockResponse.domainAssociation!.subDomains = subDomains
+    return Promise.resolve(mockResponse)
+  }
+
+  private shouldMockAmplifyCreateDomainCalls(): boolean {
+    return config.get("aws.amplify.mockAmplifyCreateDomainAssociationCalls")
+  }
+
+  private getSubDomains(
+    subDomainList: SubDomainSetting[],
+    hasCreated = false
+  ): SubDomain[] {
+    const subDomainPrefixList = subDomainList
+      .map((subDomain) => subDomain.prefix)
+      .filter((prefix) => prefix !== undefined) as string[]
+
+    const subDomains: SubDomain[] = []
+    subDomainPrefixList.forEach((subDomainPrefix) => {
+      subDomains.push({
+        dnsRecord: `${subDomainPrefix} CNAME ${
+          hasCreated ? "test.cloudfront.net" : "<pending>"
+        }`,
+        subDomainSetting: {
+          branchName: "master",
+          prefix: subDomainPrefix,
+        },
+        verified: false,
+      })
+    })
+    return subDomains
+  }
+
+  mockGetDomainAssociationOutput(
+    input: MockGetDomainAssociationCommandInput
+  ): Promise<GetDomainAssociationCommandOutput> {
+    const mockResponse: GetDomainAssociationCommandOutput = {
+      $metadata: {
+        httpStatusCode: 200,
+      },
+      domainAssociation: {
+        certificateVerificationDNSRecord: `testcert.${input.domainName}. CNAME testcert.acm-validations.aws.`,
+        domainAssociationArn: `arn:aws:amplify:ap-southeast-1:11111:apps/${input.appId}/domains/${input.domainName}`,
+        domainName: input.domainName,
+        domainStatus: "PENDING_VERIFICATION",
+        enableAutoSubDomain: false,
+        subDomains: undefined,
+        statusReason: undefined,
+      },
+    }
+    const isCreated = true // this is a get call, assume domain has already been created
+    const subDomains = this.getSubDomains(input.subDomainSettings, isCreated)
+
+    // We know that domainAssociation is not undefined, so we can use the non-null assertion operator
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    mockResponse.domainAssociation!.subDomains = subDomains
```

Comment:
we can structure this whole chunk better by placing this up top and using `subDomains` to create `mockResponse`